packages/mobile/src/store/chat/sagas.ts

```diff
@@ -8,15 +9,22 @@ const { goToChat } = chatActions
 function* watchGoToChat() {
   yield takeLatest(goToChat, function* (action: ReturnType<typeof goToChat>) {
     const {
-      payload: { chatId, presetMessage }
+      payload: { chatId, presetMessage, replaceNavigation }
     } = action
     if (navigationRef.isReady()) {
       if (!chatId) {
         // @ts-ignore navigationRef is not parametrized correctly (PAY-1141)
         navigationRef.navigate('ChatList')
       } else {
-        // @ts-ignore navigationRef is not parametrized correctly (PAY-1141)
-        navigationRef.navigate('Chat', { chatId, presetMessage })
+        console.log('goToChat', { replaceNavigation })
+        if (replaceNavigation) {
+          navigationRef.current?.dispatch(
+            StackActions.replace('Chat', { chatId, presetMessage })
```

Comment:
whoa TIL, very cool.

Reply:
Yeah it's kinda annoying but has to be a normal action dispatch because only the common actions are supported on navigationRef: https://github.com/react-navigation/react-navigation/issues/6628